### PR TITLE
feat(VExpansionPanels): Implement flat, tile, hover, and color props

### LINKED ISSUE #8939: [Feature Request] Expansion-panels not having tile, flat and color props
### Problem to solve
Not having tile, flat and color props

### Proposed solution
Should have this props like other components (ex. cards).

### PR DESCRIPTION
**Pull request by mcculleydj**
Provides styling options akin to VCard. Hover added to offer VListGroup-like styling when flat.
resolves #8939

## Description
N/A

## Motivation and Context

https://github.com/vuetifyjs/vuetify/issues/8939

## How Has This Been Tested?
Visually

## Markup:
<details>

```vue
<template>
  <v-container>
    <v-expansion-panels
      flat
      tile
      hover
    >
      <v-expansion-panel>
        <v-expansion-panel-header
          ripple
          color="primary"
        >
          My First Panel
        </v-expansion-panel-header>
        <v-expansion-panel-content color="warning">
          Lorem ipsum
        </v-expansion-panel-content>
      </v-expansion-panel>
      <v-expansion-panel>
        <v-expansion-panel-header ripple>
          My Second Panel
        </v-expansion-panel-header>
        <v-expansion-panel-content>
          Lorem ipsum
          <v-expansion-panels>
            <v-expansion-panel>
              <v-expansion-panel-header>
                My Nested Panel
              </v-expansion-panel-header>
              <v-expansion-panel-content>
                Lorem Ipsum
              </v-expansion-panel-content>
            </v-expansion-panel>
          </v-expansion-panels>
        </v-expansion-panel-content>
      </v-expansion-panel>
    </v-expansion-panels>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
    //
    }),
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)
- [ ] I've added new examples to the kitchen (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/lang/en/components/ExpansionPanels.json [hunk 1]
```diff
@@ -49,9 +49,12 @@
       "disabled": "Disables the entire expansion-panel",
       "expand": "Leaves the expansion-panel open while selecting another",
       "focusable": "Makes the expansion-panel headers focusable",
+      "flat": "Removes the expansion-panel's elevation",
```
Comment (KaelWD) [inline packages/docs/src/lang/en/components/ExpansionPanels.json]: Mention the borders too

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanels.ts [hunk 2]
```diff
@@ -24,10 +24,13 @@ export default BaseItemGroup.extend({
   props: {
     accordion: Boolean,
     disabled: Boolean,
+    flat: Boolean,
+    hover: Boolean,
```
Comment (KaelWD) [inline packages/vuetify/src/components/VExpansionPanel/VExpansionPanels.ts]: focusable already does this?

File: packages/docs/src/lang/en/components/ExpansionPanels.json (hunk 1 above)
Comment (mcculleydj) [reply packages/docs/src/lang/en/components/ExpansionPanels.json]: Good point. Pushing an update now.

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanels.ts (hunk 2 above)
Comment (mcculleydj) [reply packages/vuetify/src/components/VExpansionPanel/VExpansionPanels.ts]: I don't think so, no. Focusable goes a step further and applies a style to the header when expanded, which may or may not be desired. Hover only applies a this style on hover. If you think this prop is redundant with focusable, I'll remove it, but I would personally use hover along with flat if it were available.

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass [hunk 3]
```diff
@@ -184,3 +189,14 @@
 
     &--active
       max-width: $expansion-panel-inset-active-max-width
+
+.v-expansion-panels--flat
+  .v-expansion-panel
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass]: These styles will cascade to the inner panel, see https://github.com/vuetifyjs/vuetify/issues/9692 / https://github.com/vuetifyjs/vuetify/pull/9696

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass [hunk 4]
```diff
@@ -21,6 +21,11 @@
     .v-expansion-panel-header
       +states($material)
 
+  &.v-expansion-panels--hover
+    .v-expansion-panel-header
```
Comment (jacekkarczmarczyk) [inline packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass]: These styles will cascade to the inner panel, see https://github.com/vuetifyjs/vuetify/issues/9692 / https://github.com/vuetifyjs/vuetify/pull/9696

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass (hunk 3 above)
Comment (mcculleydj) [reply packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass]: yup, pushed an update

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass (hunk 4 above)
Comment (mcculleydj) [reply packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass]: yup, pushed an update

**Comment by johnleider** [comment]
image

Is this the desired look with tile?

**Comment by mcculleydj** [comment]
No, I think `<v-expansion-panels tile>` should square all borders for direct `v-expansion-panel` children. I pushed another commit to address this. Thanks for pointing it out.

File: packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass [hunk 5]
```diff
@@ -91,6 +97,7 @@
         transform: rotate(-180deg)
 
 .v-expansion-panels:not(.v-expansion-panels--accordion)
```
Comment (KaelWD) [inline packages/vuetify/src/components/VExpansionPanel/VExpansionPanel.sass]: I think there was supposed to be a comma here or something: 30b8b9e74

**Pull request merged**